frontend/src/components/metadata/CreateMetadataDefinition.tsx

```diff
@@ -0,0 +1,307 @@
+						</StepContent>
+					</Step>
+
+					{formInput.fields.map((input, index) => {
+						return (<Step key={index}>
+								{index == 0 ? 
+									<StepLabel>Add metadata entry*
+										{index == activeStep - 1 ? 
+										<IconButton color="primary" size="small" onClick={() => addNewField(index)}>
+											<AddBoxIcon />
+										</IconButton>
+										: <></>}
+									</StepLabel>
+								:
+									<StepLabel>Add additional entry
+										{index == activeStep - 1 ? 
+										<>
+											<IconButton color="primary" size="small" onClick={() => addNewField(index)}>
+												<AddBoxIcon />
+											</IconButton>
+											<IconButton color="primary" size="small" onClick={() => removeField(index)}>
+												<DeleteOutlineIcon />
+											</IconButton>
+										</>
+										: 
+										<IconButton size="small" onClick={() => removeField(index)}>
+											<DeleteOutlineIcon />
+										</IconButton>}
+									</StepLabel>
+								}
+								<StepContent>
+									<Grid container>
+										<Grid xs={4} md={4}>
+											<FormGroup>
+												<FormControlLabel control={<Checkbox 
+													checked={input.list}
+													onChange={(event) => { handleInputChange(index, "list", event.target.value); }}
+													/>} label="Contains List" />
+												<FormControlLabel control={<Checkbox 
+													checked={input.required}
+													onChange={(event) => { handleInputChange(index, "required", event.target.value); }}
+													/>} label="Required" />
+											</FormGroup>
+										</Grid>
+									</Grid>
+									<TextField
+										variant="outlined"
+										margin="normal"
+										required
+										fullWidth
+										id="field-name"
+										label="Field Name"
+										placeholder="Please enter field name"
+										value={input.name}
+										onChange={(event) => { handleInputChange(index, "name", event.target.value); }}
+									/>
+									<FormControl fullWidth>
+										<InputLabel id="widget-type-select">Select Widget Type</InputLabel>
+										<Select
+											labelId="widget-type-select-label"
+											id="widget-type-select-label"
+											value={input.widgetType}
+											onChange={(event) => { handleInputChange(index, "widgetType", event.target.value); }}
+											label="widgetType">
+											{widgetTypes.map((input, index) => {
+												return (
+													<MenuItem value={input}>{input}</MenuItem>
+												);
+											})}
+										</Select>
+									</FormControl>
+									<TextField
+										variant="outlined"
+										margin="normal"
+										required
+										fullWidth
+										id="datatype"
+										label="Field Data Type"
+										placeholder="Please enter field data type (e.g. int, str, enum)"
+										name="Field Data Type"
+										value={input.config.type}
+										onChange={(event) => { handleInputChange(index, "type", event.target.value); }}
```

Comment:
Since this right now is also an arbitrary list, could you also make it a dropdown?
```
"int": int,
"float": float,
"str": str,
"TextField": str,
"bool": bool,
"date": datetime.date,
"time": datetime.time,
"dict": dict,  # TODO: does this work?
"enum": str,  # TODO: need a way to validate enum,
"tuple": tuple,
```

Reply:
For both the widget type and data type drop downs, can we use more layman's term? e.g. TextField --> Text Box; Select --> Dropdown, Int --> Integer, Float --> Number
You are welcome to improvise a little bit :D

Reply:
I've added human readable names in both drop downs

<img width="1624" alt="image" src="https://user-images.githubusercontent.com/28983148/198206993-defdd3e6-bddb-40d5-986a-9e6348fa34f6.png">

<img width="1624" alt="image" src="https://user-images.githubusercontent.com/28983148/198207039-1388ce5a-0ce6-4222-8e10-e38374187faf.png">